Mathlib/Computability/Ackermann.lean

```diff
@@ -348,3 +349,52 @@ theorem not_primrec_ack_self : ¬Primrec fun n => ack n n := by
 /-- The Ackermann function is not primitive recursive. -/
 theorem not_primrec₂_ack : ¬Primrec₂ ack := fun h =>
   not_primrec_ack_self <| h.comp Primrec.id Primrec.id
+
+namespace Nat.Partrec.Code
+
+/-- The code for the partial applied Ackermann function.
+This is used to prove that the Ackermann function is computable. -/
+def pappAck : ℕ → Code
```

Comment:
These changes do make the code look nicer, though that's not quite what I meant. I was expecting something like
```
theorem primrec_ack_apply (n : ℕ) : Primrec (ack n) := sorry
```
This should then (hopefully) give you a simple proof of computability.